src/components/Message/MessageSimple/MessageActionSheet.js

```diff
@@ -0,0 +1,233 @@
+    actionSheetStyles,
+    canDeleteMessage,
+    canEditMessage,
+    handleDelete,
+    handleEdit,
+    messageActions = Object.keys(MESSAGE_ACTIONS),
+    openReactionPicker,
+    openThread,
+    options,
+    reactionsEnabled,
+    repliesEnabled,
+    setActionSheetVisible,
+    setOptions,
+    threadList,
+  } = props;
+
+  const { t } = useContext(TranslationContext);
+
+  useEffect(() => {
+    const newOptions = [];
+
+    if (
+      reactionsEnabled &&
+      messageActions.indexOf(MESSAGE_ACTIONS.reactions) > -1
+    ) {
+      newOptions.splice(1, 0, {
+        id: MESSAGE_ACTIONS.reactions,
+        title: t('Add Reaction'),
+      });
+    }
+
+    if (
+      repliesEnabled &&
+      messageActions.indexOf(MESSAGE_ACTIONS.reply) > -1 &&
+      !threadList
+    ) {
+      newOptions.splice(1, 0, { id: MESSAGE_ACTIONS.reply, title: t('Reply') });
+    }
+
+    if (messageActions.indexOf(MESSAGE_ACTIONS.edit) > -1 && canEditMessage()) {
+      newOptions.splice(1, 0, {
+        id: MESSAGE_ACTIONS.edit,
+        title: t('Edit Message'),
+      });
+    }
+
+    if (
+      messageActions.indexOf(MESSAGE_ACTIONS.delete) > -1 &&
+      canDeleteMessage()
+    ) {
+      newOptions.splice(1, 0, {
+        id: MESSAGE_ACTIONS.delete,
+        title: t('Delete Message'),
+      });
+    }
+
+    setOptions((prevOptions) => [...prevOptions, ...newOptions]);
+  }, []);
+
+  const onActionPress = async (action) => {
+    switch (action) {
+      case MESSAGE_ACTIONS.edit:
+        handleEdit();
+        break;
+      case MESSAGE_ACTIONS.delete:
+        await handleDelete();
+        break;
+      case MESSAGE_ACTIONS.reply:
+        openThread();
+        break;
+      case MESSAGE_ACTIONS.reactions:
+        openReactionPicker();
+        break;
+      default:
+        break;
+    }
+    setActionSheetVisible(false);
+  };
+
+  return (
+    <ActionSheet
+      cancelButtonIndex={0}
+      destructiveButtonIndex={0}
+      onPress={(index) => onActionPress(options[index].id)}
+      options={options.map((o, i) => {
+        if (i === 0) {
+          return (
+            <ActionSheetCancelButtonContainer testID='cancel-button'>
+              <ActionSheetCancelButtonText>
+                {t('Cancel')}
+              </ActionSheetCancelButtonText>
+            </ActionSheetCancelButtonContainer>
+          );
+        }
+        return (
+          <ActionSheetButtonContainer
+            key={o.title}
+            testID={`action-sheet-item-${o.title}`}
+          >
+            <ActionSheetButtonText>{o.title}</ActionSheetButtonText>
+          </ActionSheetButtonContainer>
+        );
+      })}
+      ref={ref}
+      styles={actionSheetStyles}
+      title={
+        <ActionSheetTitleContainer testID='action-sheet-container'>
+          <ActionSheetTitleText>{t('Choose an action')}</ActionSheetTitleText>
+        </ActionSheetTitleContainer>
+      }
+    />
+  );
+});
+
+MessageActionSheet.displayName = 'messageActionSheet';
```

Comment:
yeah something about our eslint settings throws this displayName error every so often, should prob just turn that rule off

Reply:
I think it's because Message and MessageInput components live in the same View render together and both have their own functional components called ActionSheet. If we change the naming of those the displayName may not be needed

Reply:
even when i rename, the eslint warning still shows up. it appears to have something to do with wrapping a component in React.memo/forwardRef, as i've only noticed it pop up in these components

Reply:
aight, no worries